### PR TITLE
Add data model suite

### PR DESCRIPTION
**Pull request by aljones15**
Adds a bunch of issuer tests related to the Data Model section.

### CODE REVIEW

**Review by PatStLouis — APPROVED** [review]

File: tests/suites/data-model.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+          };
+        });
+        function assertBefore() {
+          expect(
+            securedCredential,
+            `Expected issuer ${name}: ${keyType} to issue a VC`
+          ).to.exist;
+          expect(
+            securedCredential,
+            'Expected VC to be an object'
```
Comment (TallTed) [inline tests/suites/data-model.js:62]: Sometimes ``` ` ```, sometimes ` ' `? Pick one!
```suggestion
            `Expected issuer ${name}: ${keyType} to issue a VC.`
          ).to.exist;
          expect(
            securedCredential,
            'Expected VC to be an object.'
```

File: tests/suites/data-model.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+          ).to.be.an('object');
+        }
+        it('The publicKeyMultibase value of the verification method MUST ' +
+          'start with the base-58-btc prefix (z), as defined in the ' +
+          'Multibase section of Controller Documents 1.0. A ',
```
Comment (TallTed) [inline tests/suites/data-model.js:67]: I'm not sure what *should* be written here. Maybe the ` A ` should just be dropped? Maybe `1.0.A` was meant? For certain, the sentence should end with a `.`.
```suggestion
          'start with the base-58-btc prefix ("z"), as defined in the ' +
          'Multibase section of Controller Documents 1.0.A.',
```

File: tests/suites/data-model.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          assertBefore();
+          for(const proof of proofs) {
+            expect(proof.verificationMethod).to.exist;
+            expect(proof.verificationMethod).to.be.a('string');
+            const didDoc = await didResolver({url: proof.verificationMethod});
+            expect(didDoc).to.be.an('object');
+            expect(didDoc.publicKeyMultibase).to.be.a('string');
+            expect(
+              assertions.shouldBeBs58(didDoc.publicKeyMultibase),
+              'Expected "publicKeyMultibase" to be Base58 encoded'
```
Comment (TallTed) [inline tests/suites/data-model.js:79]:
```suggestion
              'Expected "publicKeyMultibase" to be Base58 encoded.'
```

File: tests/suites/data-model.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+            assertBefore();
+            for(const proof of proofs) {
+              expect(proof.verificationMethod).to.exist;
+              expect(proof.verificationMethod).to.be.a('string');
+              const didDoc = await didResolver({url: proof.verificationMethod});
+              expect(didDoc).to.be.an('object');
+              expect(didDoc.publicKeyMultibase).to.be.a('string');
+              expect(
+                assertions.shouldBeBs58(didDoc.publicKeyMultibase),
+                'Expected "publicKeyMultibase" to be Base58 encoded'
```
Comment (TallTed) [inline tests/suites/data-model.js:95]:
```suggestion
                'Expected "publicKeyMultibase" to be Base58 encoded.'
```

**Review by TallTed — CHANGES_REQUESTED** [review]

File: tests/suites/data-model.js (hunk 2 above)
Comment (aljones15) [reply tests/suites/data-model.js:67]: @TallTed this looks like a typo resulting my cut and paste job. The line reads: "The publicKeyMultibase value of the verification method MUST start with the base-58-btc prefix (z), as defined in the [Multibase section](https://www.w3.org/TR/controller-document/#multibase-0) of [Controller Documents 1.0](https://www.w3.org/TR/controller-document/)" The A belongs to the next sentence: "A Multibase-encoded ECDSA 256-bit public key value or an ECDSA 384-bit public key value follows, as defined in the [Multikey section](https://www.w3.org/TR/controller-document/#Multikey) of [Controller Documents 1.0](https://www.w3.org/TR/controller-document/). Any other encoding MUST NOT be allowed."

So the A should be removed.

File: tests/suites/data-model.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+            `Expected issuer ${name}: ${keyType} to issue a VC.`
+          ).to.exist;
+          expect(
+            securedCredential,
+            'Expected VC to be an object.'
+          ).to.be.an('object');
+        }
+        it('The publicKeyMultibase value of the verification method MUST ' +
+          'start with the base-58-btc prefix (z), as defined in the ' +
+          'Multibase section of Controller Documents 1.0. A ',
```
Comment (aljones15) [inline tests/suites/data-model.js:67]:
```suggestion
          'Multibase section of Controller Documents 1.0. ',
```

File: tests/suites/data-model.js (hunk 2 above)
Comment (aljones15) [reply tests/suites/data-model.js]: @TallTed addressed here: https://github.com/w3c/vc-di-ecdsa-test-suite/pull/95/commits/df7e5726c08069bda91aab71e0ac19af9b08ce3f

**Review by TallTed — COMMENTED** [review]
English looks OK. I don't speak JavaScript.

**Comment by aljones15** [comment]
> English looks OK. I don't speak JavaScript.

@TallTed you should learn. It's a great language.

**Comment by TallTed** [comment]
@aljones15 

> you should learn. It's a great language.

For various reasons, I prefer to keep English as my primary coding language, with which I achieve reasonably good results, even with my imperfect facility.

**Pull request merged**